p2p_wallet/Scenes/Main/Swap/State/JupiterSwapStateMachine.swift

```diff
@@ -1,21 +1,68 @@
-import Send // TODO: I will extract StateMachine in core module inside Key App Kit
 import Combine
 
-actor JupiterSwapStateMachine: StateMachine {
+actor JupiterSwapStateMachine {
+    // MARK: - Nested type
+    
+    /// The cache that handle currentTask and currentAction
+    /// Must be actor to make sure that currentTask and currentAction are thread-safe
+    private actor Cache {
+        /// Current executing task
+        fileprivate var currentTask: Task<JupiterSwapState, Never>?
+        /// Save the current task
+        fileprivate func saveCurrentTask(_ task: Task<JupiterSwapState, Never>?) {
+            currentTask = task
+        }
+    }
```

Comment:
Add `Cache` to manage currentTask (ex:. canceling).
Cache type is `Actor` to make writing/reading currentTask safe from other asynchronous context.